scripts/lib/CIME/SystemTests/erp.py

```diff
@@ -8,132 +8,72 @@
-            stop_option = self._case.get_value("STOP_OPTION")
-
-            if run == 1:
-                expect(stop_n > 2, "ERROR: stop_n value {:d} too short".format(stop_n))
-                rest_n = stop_n/2 + 1
-                self._case.set_value("REST_N", rest_n)
-                self._case.set_value("REST_OPTION", stop_option)
-                self._case.set_value("HIST_N", stop_n)
-                self._case.set_value("HIST_OPTION", stop_option)
-                self._case.set_value("CONTINUE_RUN", False)
-                suffix = "base"
-            else:
-                rest_n = stop_n/2 + 1
-                stop_new = stop_n - rest_n
-                expect(stop_new > 0, "ERROR: stop_n value {:d} too short {:d} {:d}".format(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-
-            case_setup(self._case, test_mode=True, reset=True)
-
-            self.run_indv(suffix=suffix)
 
-        self._component_compare_test("base", "rest")
+    def _case_one_setup(self):
+        stop_n      = self._case.get_value("STOP_N")
+
+        expect(stop_n > 2, "ERROR: stop_n value {:d} too short".format(stop_n))
+
+    def _case_two_setup(self):
+        # halve the number of tasks and threads
+        for comp in self._case.get_values("COMP_CLASSES"):
+            ntasks    = self._case1.get_value("NTASKS_{}".format(comp))
+            nthreads  = self._case1.get_value("NTHRDS_{}".format(comp))
+            rootpe    = self._case1.get_value("ROOTPE_{}".format(comp))
+            if ( nthreads > 1 ):
+                self._case.set_value("NTHRDS_{}".format(comp), nthreads/2)
+            if ( ntasks > 1 ):
+                self._case.set_value("NTASKS_{}".format(comp), ntasks/2)
+                self._case.set_value("ROOTPE_{}".format(comp), rootpe/2)
+
+        stop_n = self._case1.get_value("STOP_N")
+        rest_n = self._case1.get_value("REST_N")
+        stop_new = stop_n - rest_n
+        expect(stop_new > 0, "ERROR: stop_n value {:d} too short {:d} {:d}".format(stop_new,stop_n,rest_n))
+        self._case.set_value("STOP_N", stop_new)
+        self._case.set_value("HIST_N", stop_n)
+        self._case.set_value("CONTINUE_RUN", True)
+        self._case.set_value("REST_OPTION","never")
+
+        # Note, some components, like CESM-CICE, have
+        # decomposition information in env_build.xml that
+        # needs to be regenerated for the above new tasks and thread counts
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _case_one_custom_postrun_action(self):
+        rundir1 = self._case1.get_value("RUNDIR")
+        rundir2 = self._case2.get_value("RUNDIR")
+        case = self._case1.get_value("CASE")
+        datenames = _get_datenames(self._case1)
+        for file_ in glob.iglob(os.path.join(rundir1,"*")):
```

Comment:
This copies the rpointer files and links the restart and hist restart files from the restart time to the case2 run directory.

Reply:
Thank you for coming up with this solution that avoids running the full short-term archiver.

Could the bulk of this function be put in some shared location so that it can be reused by other tests if needed? This could go in `scripts/lib/CIME/SystemTests/test_utils/`.